### PR TITLE
Add gisf command-line option for guiserver

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/95/
Test PASSed.

File: server/guiserver/manage.py [hunk 1]
```diff
@@ -161,6 +161,7 @@ def setup():
         'gzip', type=bool, default=False,
         help='Enable gzip compression in the gui.')
     define('gtm', type=bool, default=False, help='Enable Google tag manager.')
+    define('gisf', type=bool, default=False, help='Enable GUI in store front.')
```
Comment (hatched) [inline server/guiserver/manage.py]: Why did this cause the charm to not deploy? If this is so critical shouldn't we have a test for it?

**Comment by bac** [comment]
QA:

deploy charm
ensure you can hit the GUI in the browser
`juju run --unit juju-gui/0 'cat /usr/local/bin/runserver.sh'`
ensure no gisf
`juju set-config juju-gui gisf-enabled=true`
`juju run --unit juju-gui/0 'cat /usr/local/bin/runserver.sh'`
ensure --gisf

**Comment by jaycee** [comment]
:+1: 

I'm wondering how this got past QA previously, since that should have caught this issue. Did we accidentally not QA your previous PR?

This definitely supports the idea that we need to rethink the process of adding options to the gui charm; there's a lot of possible break points.

**Comment by jaycee** [comment]
QAing now.

**Comment by makyo** [comment]
:+1:

**Comment by jaycee** [comment]
QA OK.

**Comment by jaycee** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Comment by bac** [comment]
Fixed Issue #53

**Comment by bac** [comment]
The QA on the previous branch was incomplete. It focused on seeing that the option was written to runserver.sh but didn't actually hit the GUI to see that it had started. 

Yes @hatched we need a suite of tests that ensure the options we are setting are actually accepted.

**Pull request merged**